reactor-core/src/main/java/reactor/core/scheduler/SchedulerMetricDecorator.java

```diff
@@ -76,13 +84,12 @@ in order to avoid consider two calls in a row as duplicates (yet still being abl
 		to distinguish between two instances with the same name and configuration).
 		 */
 
+		ExecutorServiceMetrics.monitor(globalRegistry, service, executorId, tags);
```

Comment:
this method does 2 things, only 1 of which we're interested in:

```java
new ExecutorServiceMetrics(executor, executorServiceName, tags).bindTo(registry);
return new TimedExecutorService(registry, executor, executorServiceName, tags);
```

Since the second half is basically ignored and replaced by the `new TimedScheduledExecutorService` call below, we could replace the call to `monitor` with:

```java
new ExecutorServiceMetrics(executor, executorServiceName, tags).bindTo(registry);
```

Reply:
that said, calling `monitor` is the "recommended" way and additional useful behavior could be introduced in there later on...